packages/permissions/src/hooks/use-is-authorized/use-is-authorized.ts

```diff
@@ -94,14 +86,17 @@ const useIsAuthorized = ({
     )}.`
   );
 
-  const actualPermissions = useApplicationContext<TPermissions | null>(
-    (applicationContext) => applicationContext.permissions
+  const actualPermissions = useApplicationContext<TNormalizedPermissions | null>(
+    (applicationContext) =>
+      projectPermissions?.permissions ?? applicationContext.permissions
   );
-  const actualActionRights = useApplicationContext<TActionRights | null>(
-    (applicationContext) => applicationContext.actionRights
+  const actualActionRights = useApplicationContext<TNormalizedActionRights | null>(
+    (applicationContext) =>
+      projectPermissions?.actionRights ?? applicationContext.actionRights
   );
-  const actualDataFences = useApplicationContext<TDataFences | null>(
-    (applicationContext) => applicationContext.dataFences
+  const actualDataFences = useApplicationContext<TNormalizedDataFences | null>(
+    (applicationContext) =>
+      projectPermissions?.dataFences ?? applicationContext.dataFences
```

Comment:
As you can see here, we try to read/resolve the permissions passed as prop first, otherwise falling back to the application context.